### PR TITLE
chore(server): remove unnecessary promise from getConfig

### PR DESCRIPTION
**Pull request by lmiller1990**
- Closes N/A

### User facing changelog

`getConfig` no longer has any async behavior, so I removed `async` and updated the places where we did `getConfig().then` to just do `const cfg = getConfig()`. I'd recommend reviewing with white space off.

### Additional details

Remove unnecessary promise. It basically simplifies things a bit, and removes the need to call `then` when you want to get the config.

### PR Tasks

- [x] Have tests been added/updated?
- [ ] Has the original issue or this PR been tagged with a release in ZenHub? 
- [ ] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [ ] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
- [ ] Have new configuration options been added to the [`cypress.schema.json`](https://github.com/cypress-io/cypress/blob/develop/cli/schema/cypress.schema.json)?

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
@lmiller1990 Some lint errors

File: packages/server/lib/gui/files.ts [hunk 1]
```diff
@@ -2,40 +2,30 @@ import openProject from '../open_project'
 import { createFile } from '../util/spec_writer'
 import { showSaveDialog } from './dialog'
 
-export const showDialogAndCreateSpec = () => {
```
Comment (lmiller1990) [inline packages/server/lib/gui/files.ts:5]: `getConfig` no longer returns a promise..., so no more `then`, and no more `tap`, so I just made it into a regular old function, replacing `then` with `await`.

File: packages/server/lib/open_project.js [hunk 2]
```diff
@@ -84,101 +84,100 @@ const moduleFactory = () => {
...
 
-        // if we don't have the isHeaded property
-        // then we're in interactive mode and we
-        // can assume its a headed browser
-        // TODO: we should clean this up
-        if (!_.has(browser, 'isHeaded')) {
-          browser.isHeaded = true
-          browser.isHeadless = false
-        }
+      _.defaults(options, {
```
Comment (lmiller1990) [inline packages/server/lib/open_project.js:89]: this entire chunk is the same, removing the promise removed `then` and a ton of white space which makes the code look different

**Comment by lmiller1990** [comment]
@jennifer-shehane thanks, I have fixed the linting errors and merged in develop ✅

File: packages/server/lib/gui/files.ts [hunk 3]
```diff
@@ -2,40 +2,30 @@ import openProject from '../open_project'
...
+  if (!path) {
+    return {
+      specs: null,
+      path,
     }
+  }
+
+  // only create file if they selected a file
+  if (path) {
+    createFile(path)
```
Comment (chrisbreiding) [inline packages/server/lib/gui/files.ts:19]: `createFile` returns a promise, so this should be awaited, right?

**Review by chrisbreiding — CHANGES_REQUESTED** [review]
Very nice - I love how this cleans things up. Just the one thing to address, but otherwise looks great.

File: packages/server/lib/gui/files.ts (hunk 3 above)
Comment (lmiller1990) [reply packages/server/lib/gui/files.ts:19]: Right, good catch - I forgot that `fs.writeFile` is actually wrapped with promisify.

**Comment by lmiller1990** [comment]
@chrisbreiding I made the change to `await fs.writeFile`.

**Review by chrisbreiding — APPROVED** [review]

**Pull request merged**